.circleci/tests/test_conn.py

```diff
@@ -0,0 +1,159 @@
+    atlas = os.path.join(fcon_ts_wf.base_dir,
+                         "fcons_ts_wf/apply_"
+                         "transform_schaefer_417/", atlas)
+    atlas = nilearn.image.load_img(atlas)
+    # Masking img
+    masker = NiftiLabelsMasker(atlas, standardize=False)
+    # Fitting mask
+    masker.fit(fake_bold_file)
+    signals = masker.transform(fake_bold_file)
+    # The "ground truth" matrix
+    ground_truth = np.corrcoef(signals.T)
+    assert (np.sum(abs(ground_truth - xcp_array))) < 0.01
+    assert (np.max(abs(ground_truth - xcp_array))) < 0.01
+
+    return
+
+
+def cifti_con_test(data_dir, tmp_path_factory):
+    """Test the cifti workflow - only correlation, not parcellation."""
+    # Define bold file
+    boldfile = os.path.join(
+        data_dir, "fmriprep/sub-colornest001/ses-1/func",
+        "sub-colornest001_ses-1_task-rest_run-2_space-",
+        "fsLR_den-91k_bold.dtseries.nii"
+    )
+    # Generate fake signal
+    bold_data = read_ndata(boldfile)
+    shape = bold_data.shape  # get the shape so we can generate a
+    # matrix of random numbers with the same shape
+    fake_signal = np.random.randint(bold_data.min(),
+                                    bold_data.max(),
+                                    size=shape)
+    # Let's write that out
+    tmpdir = tmp_path_factory.mktemp("fcon_cifti_test")
+    filename = os.path.join(tmpdir, "fake_signal_file.dtseries.nii")
+    write_ndata(
+        fake_signal,
+        template=boldfile,
+        TR=_get_tr(nb.load(boldfile)),
+        filename=filename
+    )
+    fake_bold_file = filename
+    # Create the node and a tempdir to write its results out to
+    tmpdir = tmp_path_factory.mktemp("fcon_cifti_test_2")
+    cifti_conts_wf = init_cifti_conts_wf(
+        mem_gb=4, name="cifti_ts_con_wf", omp_nthreads=2
+    )
+    cifti_conts_wf.base_dir = tmpdir
+    # Run the node
+    cifti_conts_wf.inputs.inputnode.clean_cifti = fake_bold_file
+    cifti_conts_wf.run()
+    # Let's find the correct parcellated file
+    for file in os.listdir(cifti_conts_wf.base_dir + "cifti_ts_con_wf/sc417parcel"):
+        if fnmatch.fnmatch(file, "*dtseries*"):
+            out_file = file
+    out_file = os.path.join(cifti_conts_wf.base_dir,
+                            "cifti_ts_con_wf/sc417parcel", out_file)
+    # Let's read out the parcellated time series and get its corr coeff
+    data = read_ndata(out_file)
+    ground_truth = np.corrcoef(data)
+    # Let's find the conn matt generated by XCP
+    for file in os.listdir(cifti_conts_wf.base_dir + "/cifti_ts_con_wf/sc417corr"):
+        if fnmatch.fnmatch(file, "*matrix*"):
+            out_file = file
+    out_file = os.path.join(cifti_conts_wf.base_dir,
+                            "cifti_ts_con_wf/sc417corr", out_file)
+    # Read it out
+    data = read_ndata(out_file)
+    # Do the two match up?
+    assert (np.max(np.abs(data - ground_truth))) < 0.01
+    assert (np.sum(np.abs(data - ground_truth))) < 0.01
```

Comment:
If these are arrays, I'd prefer to use `numpy.allclose`, with an atol value based on the accuracy we expect (0.01 it seems).